### PR TITLE
Dashboard: Responsive Style Updates

### LINKED ISSUE #1090: Dashboard: Initial updates for responsive views
After friday's demos we need to circle back to add some responsiveness to the dashboard that was not clear in the figma guides. 

Audit and fix the layout to look good on mobile devices as well so that we can have that moving forward for the other views. It looks like the figma designs only really handled up until tablet, but we have enough context to ensure a solid mobile view.

Areas to look into are

Header (nav)
Heading (search)
Font sizes (heading)
Grid should default to 2 columns per row 
Landscape and Portrait views

### PR DESCRIPTION
**Pull request by BrittanyIRL**
# Overview
- cleaning up smaller viewports of dashboard 

## Before
Screen Shot 2020-04-07 at 8 32 49 AM

- Figma had led me to assume that we were just working through tablet size. This PR is meant to remedy that initial error. I took some liberties that I will discuss with @samitron7 - in places where I alter from our given designs it's for space allocation and to start a conversation.  

## Issues addressed 
- Overall
  - added in 1 more breakpoint at 440px to handle most phone sizes 
  - added gutter to theme, desktop and min (20px/ 10px) 
  - removed that left padding of 10px that we were getting by default from styles.css. The top level classname was wrong so those style overrides weren't happening. 

- Navigation (top most section) 
  - As you shrink past the dropdown version, you'll get the create story button to the top right and any navigation links below that - since the dashboard only has 3 I thought just having them exposed was an easier way to move about and cleaner from a component standard. Right now, they are hidden because we have 1 page, which is a redundant use of space. So see this in action, uncomment the links in our constants.js file 

- Heading
  - This is the section I'm least sure about. I have moved the search component below the page title / search results heading when relevant. Did this so that longer search result titles could take up room and the search component when expanded wouldn't eat up the text or be some weird width. 
  - Bit smaller font for heading
  - Some padding condense 

- Page filters 
  - When you hit that 440px media query, removing the 'pill' styling for 'all stories', 'drafts', 'active stories' to remove some excess space and keep them all on one row 

- Grid 
  - Forcing 2 columns per row at 440px. 
  - Smaller viewports get a centered grid
  - Added in 1 more grid item size for 440px media query 
  - grid preview container updates according to that 

## Testing 
- Resize the dashboard, see how it works and tell me if I missed something! 

### Screenshots 
Screen Shot 2020-04-07 at 3 19 38 PM

Screen Shot 2020-04-07 at 3 19 50 PM

Screen Shot 2020-04-07 at 3 19 58 PM

Screen Shot 2020-04-07 at 3 20 09 PM

### Demos
 
**Iphone SE - horizontal**
iphone se horizontal

**Iphone 8 - vertical**
iphone8

### CODE REVIEW

File: assets/src/dashboard/app/views/myStories/index.js [hunk 1]
```diff
@@ -41,12 +41,31 @@ import NoResults from './noResults';
...
+      box-shadow: none;
+      padding: 0 10px 0 0;
+    }
+  }
+`;
+const BodyWrapper = styled.div`
+  margin: ${({ theme }) => `0 ${theme.pageGutter.desktop}px`};
+
+  @media ${({ theme }) => theme.breakpoint.smallDisplayPhone} {
+    margin: ${({ theme }) => `0 ${theme.pageGutter.min}px`};
```
Comment (BrittanyIRL) [inline assets/src/dashboard/app/views/myStories/index.js:63]: i didn't feel like we needed 5 pageGutters, so i just did desktop and min, which is a true min. How do you both feel about intermingling the word 'min' like this?

**Review by mariano-formidable — CHANGES_REQUESTED** [review]
Hmmmm... I'm noticing that we're setting `width` and `height` in a lot of places using similar (or the same) breakpoints, _and_ that we have the `usePagePreviewSize` which also calculates the appropriate `width` and `height` of a card based on the window size... I'm curious if there's a way to use `usePagePreviewSize` to set the `width` and `height` for _all the things_ that need it, and then we can remove a bunch of the media queries we have that implement the same restriction.

I only ask because right now it's possible for our breakpoints and `usePagePreviewSize` to get out of sync.  In fact, I think I found an out-of-sync bug in the current code, in the gif below I'm highlighting the `PreviewPane` container and the `DisplayElement` container and notice how the `PreviewPane` container's width & height (set by media queries) is smaller than the `DisplayElement` container (set by `usePagePreviewSize`).

iphone_grid

Refactoring all the subcomponents to use `usePagePreviewSize` instead of media queries would be a bit of work, and I know you've already put in a lot in this refactor sooooo if it doesn't make sense feel free to push back!

**Comment by carloskelly13** [comment]
I agree with Mariano that we likely have to leverage `usePagePreviewSize` since it's the source of truth. But also we will need flexible height for the title and date. Since those can extend a couple of lines depending on the width. I do think it's fine to make that a future PR since this is quite large.

**Comment by carloskelly13** [comment]
For mobile I do wonder if it would be better to do the page dropdown + search on the same line.

**Review by carloskelly13 — APPROVED** [review]
I think this moves us in the right direction absent designs, but I do agree with Mariano that we should find a way to leverage the hook that calculates the story width.

**Comment by BrittanyIRL** [comment]
> Hmmmm... I'm noticing that we're setting `width` and `height` in a lot of places using similar (or the same) breakpoints, _and_ that we have the `usePagePreviewSize` which also calculates the appropriate `width` and `height` of a card based on the window size... I'm curious if there's a way to use `usePagePreviewSize` to set the `width` and `height` for _all the things_ that need it, and then we can remove a bunch of the media queries we have that implement the same restriction.
> 
> I only ask because right now it's possible for our breakpoints and `usePagePreviewSize` to get out of sync. In fact, I think I found an out-of-sync bug in the current code, in the gif below I'm highlighting the `PreviewPane` container and the `DisplayElement` container and notice how the `PreviewPane` container's width & height (set by media queries) is smaller than the `DisplayElement` container (set by `usePagePreviewSize`).
> 
> iphone_grid
> 
> Refactoring all the subcomponents to use `usePagePreviewSize` instead of media queries would be a bit of work, and I know you've already put in a lot in this refactor sooooo if it doesn't make sense feel free to push back!

What a good find, this rules! I am going to push back just for the sake of review time, clarity, and small(er) PRs and do this as a follow up. It should be done sooner rather than later for sure and I feel like all this clean up is only going to make the other grid views go faster.  That cool with you?

**Review by mariano-formidable — APPROVED** [review]
Great move to resize goodness!

**Pull request merged**